### DISCUSSION TITLE
Using llama as Python interpreter

### DISCUSSION THREAD

**Discussion by niansa** [ggml-org/llama.cpp#396] [Show and tell]
I experimented with using Llama to "Interpret" code:

```
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print("Hello world!")
Hello world!
>>> import huge_dictionary as dic
>>> res = dic.find("treehouse")
>>> res.description
'A treehouse is a structure built around, in, or on the trunk of a mature tree that can be used for recreation, leisure, work, or residence.'
>>>
```
It's working wonderfully! You can litterally just think of any module and it'll just... exist... and have all the functions you tell it to invoke. Pretty cool. My prompt is everything up to the `>>>` after the `Hello world!` output.

**Comment by Green-Sky** [comment]
what model did you use? this looks very good.

**Reply by niansa** [reply]
13B

**Reply by SpeedyCraftah** [reply]
I swear the models with the lower amounts of parameters shine here as opposed to 30B+
